File: solutions/scala/armstrong-numbers/1/src/main/scala/ArmstrongNumbers.scala
object ArmstrongNumbers {
  private val tolerance: Double = 1e-323

  def isArmstrongNumber(number: Int): Boolean =
    math.abs(number - sumOfDigitsToPowerOfLength(number)) < tolerance

  private def sumOfDigitsToPowerOfLength(number: Int): Double = {
    val numberString = number.toString
    numberString
      .toCharArray
      .map(x => sumOfDigitsToPowerOfLength(x, numberString.length))
      .sum
  }

  private def sumOfDigitsToPowerOfLength(x: Char, power: Int): Double =
    math.pow(x.toString.toInt, power)
}

File: solutions/csharp/armstrong-numbers/1/ArmstrongNumbers.cs
using System;
using System.Linq;

public static class ArmstrongNumbers
{
    public static bool IsArmstrongNumber(int number) =>
        number == SumOfDigitsToPowerOfLength(number);

    private static double SumOfDigitsToPowerOfLength(int number)
    {
        string numberString = number.ToString();
        return numberString.ToCharArray()
            .ToList()
            .Select(x => SumOfDigitsToPowerOfLength(x, numberString.Count()))
            .Sum();
    }

    private static double SumOfDigitsToPowerOfLength(char x, int power) =>
        Math.Pow(int.Parse(x.ToString()), power);
}